<launch>

    <arg name="camera_depth_ns" default="camera/aligned_depth_to_color" />

    <arg name="detections_topic" default="detectnet_nodelet/detections" />

    <arg name="external_manager" default="false" />
    <arg name="manager" default="nodelet_manager" />


    <node unless="$(arg external_manager)" pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" required="true" />

    <node name="target_locator_nodelet" pkg="nodelet" type="nodelet" args="load xiaoche/target_locator $(arg manager)" required="true">

        <remap from="~detections" to="$(arg detections_topic)" />
        <remap from="~depth_image_raw" to="$(arg camera_depth_ns)/image_raw" />
        <remap from="~depth_camera_info" to="$(arg camera_depth_ns)/camera_info" />

        <!-- How long do we wait until message of type sensor_msgs/CameraInfo is received -->
        <param name="camera_info_timeout" value="5.0" />

<!--        <param name="image_transport" value="compressed"/> -->

        <!-- Used by message_filters, to synchronize messages on different topics -->
        <param name="lower_bound_ms" value="0"/>
        <param name="age_penalty" value="1.0"/>

        <!-- How many percent of the bbox is used to calculate average distance -->
        <param name="horizontal_rate" value="0.8" />
        <param name="vertical_rate" value="0.8" />

    </node>


</launch>
